src/stanc/stanc.ml

```diff
@@ -306,13 +310,35 @@ let use_file filename =
     Warnings.pp_warnings Fmt.stderr ?printed_filename
       (Deprecation_analysis.collect_warnings typed_ast) ;
   if !generate_data then
-    print_endline
-      (Debug_data_generation.gen_values_json
-         (Ast_to_Mir.gather_declarations typed_ast.datablock) ) ;
-  if !generate_inits then
-    print_endline
-      (Debug_data_generation.gen_values_json
-         (Ast_to_Mir.gather_declarations typed_ast.parametersblock) ) ;
+    let decls = Ast_to_Mir.gather_declarations typed_ast.datablock in
+    let data =
+      match !data_file with
+      | None -> Map.Poly.empty
+      | Some file ->
+          Debug_data_generation.json_to_mir decls (Yojson.Basic.from_file file)
+    in
+    match Debug_data_generation.gen_values_json ~data decls with
+    | Ok s -> print_endline s
+    | Error e -> Errors.pp Fmt.stderr ?printed_filename (Errors.DebugDataError e)
+  else if !generate_inits then (
+    let data =
+      match !data_file with
+      | None -> Map.Poly.empty
+      | Some file ->
+          Debug_data_generation.json_to_mir
+            (Ast_to_Mir.gather_declarations typed_ast.datablock)
+            (Yojson.Basic.from_file file) in
+    match
+      Debug_data_generation.gen_values_json ~filter:true ~data
+        (Ast_to_Mir.gather_declarations typed_ast.parametersblock)
+    with
+    | Ok s -> print_endline s
+    | Error e ->
+        Errors.pp Fmt.stderr ?printed_filename (Errors.DebugDataError e) ;
```

Comment:
We should exit with a nonzero code here IMO

Reply:
Should we also exit with zero code in the `Ok` branch? Right now we always generate a `.hpp` file but that seems weird to me...

Reply:
> similar to how it currently functions with `--auto-format`
Oh, neat. Done.